Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go

```diff
@@ -534,10 +534,20 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 		}()
 	}
 	wg.Wait()
+
 	select {
 	case err := <-errs:
 		return nil, err
 	default:
+		anyLabels := listOptions.LabelSelector == nil || listOptions.LabelSelector.Empty()
+		anyFields := listOptions.FieldSelector == nil || listOptions.FieldSelector.Empty()
+
+		if anyLabels && anyFields && api.NamespaceValue(ctx) != "" {
+			// Try to delete the directory, which will only succeed if the directory is empty.
```

Comment:
@liggitt How many resources are subject to this graceful deletion? And in that case, don't we just have a few extra empty directories left laying around? I think as long as we do a reasonable job cleaning up as much as we can, it's better than not having this change.

Similarly, restricting this to only happen when neither label nor field selectors are specified will miss some empty etcd directories as well.

I'm not clear on what we more we need to do to proceed here - can you elaborate? cc @abhgupta